### PR TITLE
Allow HTTP caller to respond `error`

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose
$Subject

> Fixes [`Add support to respond error type using http:Caller #2832`](https://github.com/ballerina-platform/ballerina-standard-library/issues/2832)

## Examples

```ballerina
import ballerina/http;

service on new http:Listener(9090) {

    resource function get respondedError(http:Caller caller) returns error? {
        check caller->respond(error("respond with new error"));
    }

    resource function get respondedHttpError(@http:CallerInfo{respondType : http:Error} 
              http:Caller caller) returns error? {
        http:Error err = error("respond with new error");
        check caller->respond(err);
    }
}
```

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec

### CODE REVIEW

File: compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/service.bal [hunk 1]
```diff
@@ -111,4 +113,20 @@ service http:Service on new http:Listener(9090) {
...
+
+    resource function get callerInfo20(@http:CallerInfo {respondType: http:Error} http:Caller abc) returns error? {
+        check abc->respond(<http:Error> error("hello world"));
+    }
+
+    resource function get callerInfo21(@http:CallerInfo {respondType: http:Error} http:Caller abc) returns error? {
+        check abc->respond(error("hello world")); // error
+    }
+
+    resource function get greeting(@http:CallerInfo{respondType: NewError} http:Caller caller) returns error? {
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/service.bal:129]: Add a test with `{respondType: error}` as negative case. So that once this is allowed from lang team, we can turn it to a success test

Comment (TharmiganK) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/service.bal:129]: Added in [67efe99](https://github.com/ballerina-platform/module-ballerina-http/pull/960/commits/67efe99cd1f36009b67c5cb2e4895c341f0bf0f5)

**Review by chamil321 — APPROVED** [review]
